src/Text/Pandoc/Readers/Docx/Parse.hs

```diff
@@ -250,12 +253,37 @@ newtype TblLook = TblLook {firstRowFormatting::Bool}
 defaultTblLook :: TblLook
 defaultTblLook = TblLook{firstRowFormatting = False}
 
-newtype Row = Row [Cell]
-           deriving Show
+data Row = Row TblHeader [Cell] deriving Show
+
+data TblHeader = HasTblHeader | NoTblHeader deriving (Show, Eq)
 
-newtype Cell = Cell [BodyPart]
+data Cell = Cell GridSpan VMerge [BodyPart]
             deriving Show
 
+type GridSpan = Integer
+
+data VMerge = Continue
+            -- ^ This cell should be merged with the one above it
+            | Restart
+            -- ^ This cell should not be merged with the one above it
+            deriving Show
+
+rowsToRowspans :: [Row] -> [[Int]]
+rowsToRowspans rows = transpose (fmap colspans (columns rows))
+  where
+    columns :: [Row] -> [[Cell]]
+    columns = transpose . fmap (\(Row _ cs) -> cs)
+
+    colspans :: [Cell] -> [Int]
+    colspans = fmap snd . fst . foldr calculateColspan ([], 0)
+
+    -- TODO: Should the contents of merged cells be combined?
+    calculateColspan :: Cell -> ([(Cell, Int)], Int) -> ([(Cell, Int)], Int)
+    calculateColspan cell@(Cell _ vMerge _) (column, continuesBelow) =
+      case vMerge of
+        Restart -> ((cell, 1 + continuesBelow) : column, 0)
+        Continue -> (column, 1 + continuesBelow)
```

Comment:
I'm a little confused on the naming here. When you have `Restart` and `Continue`, it seems like those are for `rowspan`s, but this is named `calculateColspan`.

Reply:
Yes, the name is incorrect. When I wrote it I had colspans and rowspans backwards in my head, and when I renamed everything I missed the names inside this function.